jdaviz/configs/imviz/helper.py

```diff
@@ -144,36 +152,33 @@ def load_data(self, data, data_label=None, show_in_viewer=True, **kwargs):
         image as Numpy array and load the latter instead.
 
         """
-        prev_data_labels = self.app.data_collection.labels
+        self.app.state.dev_loaders = True
+
+        extensions = kwargs.pop('ext', None)
 
         if 'gwcs_to_fits_sip' not in kwargs and 'Orientation' in self.plugins.keys():
             kwargs['gwcs_to_fits_sip'] = self.plugins['Orientation'].gwcs_to_fits_sip
 
-        if isinstance(data, str):
-            filelist = data.split(',')
+        if isinstance(data, str) and "," in data:
+            data = data.split(',')
 
-            if len(filelist) > 1 and data_label:
+        if isinstance(data, (tuple, list)) and not isinstance(data, fits.HDUList):
+            if len(data) > 1 and data_label:
                 raise ValueError('Do not manually overwrite data_label for '
                                  'a list of images')
+            with self.batch_load():
+                for data_i in data:
+                    kw = deepcopy(kwargs)
+                    self.load_data(data_i,
```

Comment:
just curious why this is going through app.load_data, rather than self._load, as its done elsewhere?

Reply:
loaders don't (yet) have support for multiple inputs, so we need to loop over the input and make multiple calls.  This is just recursively calling itself so that the same backwards-compatibility logic can be re-used for each iteration of the loop.